driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -722,12 +759,30 @@ public boolean onTimeout(Connection connection, long latency, int retryCount) {
             }
 
             Host queriedHost = current;
-            OperationTimedOutException timeoutException = new OperationTimedOutException(connection.address);
+
+            // This cannot be OperationTimedOutException because
```

Comment:
Previously (before JAVA-698) it used to be a DriverException. The newbie I was back then (maybe I'm still am) found it weird because one would create an OperationTimedOutException instead in some other similar places (Connection.Future.onTimeout and DefaultResultSetFuture.onTimeout).
But I realized since then that the difference is that RequestHandler logs those errors and adds them to the final list of errors in NHAE. So it strives to hide non-public exceptions.
Things became complex with the QueryLogger, because some "internal" exceptions like this one would appear in client code for the first time. And now with the ExtendedRetryPolicy, users can even catch this exception if the policy says RETHROW.
(Curiously, the same logic would apply to ConnectionException, but oddly this one has never been hidden and appears bare in NHAE, Query Logger, etc. – see RequestHandler.onException.)
All of his is a bit flawed, but hopefully we can address this in a better way with JAVA-1006. I think that now that these exceptions bubble up into client code, they deserve to become public.